src/ra_server_proc.erl

```diff
@@ -385,11 +398,13 @@ leader({call, From}, {log_fold, Fun, Term}, State) ->
     fold_log(From, Fun, Term, State);
 leader(EventType, Msg, State0) ->
     case handle_leader(Msg, State0) of
-        {leader, State1, Effects} ->
-            {State, Actions} = ?HANDLE_EFFECTS(Effects, EventType, State1),
+        {leader, State1, Effects1} ->
```

Comment:
why the rename?

Reply:
To have the same number in state and events on each step. It can be hard to track numbers separately.

Reply:
Generally (but not consistently) the last assignment should be number less. So the first incarnation use the 0 suffix which is incremented until the very last incarnation which has no number. It's just an unofficial pattern that emerged but it makes it easy to see which one is the final state version.